src/metabase/events/recent_views.clj

```diff
@@ -12,11 +12,29 @@
 
 (derive ::event :metabase/event)
 
-(derive :event/dashboard-read ::event)
-(derive :event/table-read ::event)
+(derive ::dashboard-read :metabase/event)
+(derive :event/dashboard-read ::dashboard-read)
 
-(m/defmethod events/publish-event! ::event
+(m/defmethod events/publish-event! ::dashboard-read
   "Handle processing for a single event notification which should update the recent views for a user."
+  [topic {:keys [object-id user-id] :as _event}]
+  (span/with-span!
+    {:name (str "recent-views-" (name topic))
+     :topic topic
+     :user-id user-id}
+    (try
+      (let [model    (audit-log/model-name :model/Dashboard)
+            model-id object-id
```

Comment:
There's enough non-trivial duplication between these two handlers to justify a shared handler IMHO, something like `(publish-read-event! topic model model-id user-id)`. Despite being right next to each other, there's still a chance we update one and forget the other.

Reply:
Good suggestion, I decided to tackle the duplication with a different approach though. [Here's what I have so far.](960d296c94bc59186b9f72fabf27a9969c612170) 

- I removed the `span/with-span!` calls since they are mostly redundant with this call in `recent-views/update-users-recent-views!`
https://github.com/metabase/metabase/blob/960d296c94bc59186b9f72fabf27a9969c612170/src/metabase/models/recent_views.clj#L112-L116
- I de-duplicated the try/catch calls which I assume have one purpose across all events: to reduce the impact of issues with `recent-views/update-users-recent-views!`

Nothing else can really be de-duplicated since the shape of the event map differs between events.

Reply:
If you like what I did with the try/catch I'll extend this to more event types e.g. the ones in `metabase.events.view-log`

Reply:
Actually in [this PR](https://github.com/metabase/metabase/pull/43167) I'm removing this whole file, so I'll continue this refactoring there

Reply:
I've reverted the changes above in this PR 62cd24ec858882e6b7585c0d2692645e6d1eabf3